src/gui/dialogs/preferences_dialog.cpp

```diff
@@ -95,6 +97,19 @@ struct hotkey_sort_by_desc
 	}
 	const gui2::tpreferences::t_visible_hotkeys* hotkey_commands_;
 };
+
+template<bool reverse>
+struct hotkey_sort_by_command
+{
+	hotkey_sort_by_command(const gui2::tpreferences::t_visible_hotkeys& l) : hotkey_commands_(&l) {}
+	bool operator()(int lhs, int rhs) const
+	{
+		return reverse ? hotkey::get_names((*hotkey_commands_)[lhs]->command) < hotkey::get_names((*hotkey_commands_)[rhs]->command)
```

Comment:
I admit I'm not that strong on templates and STL as I didn't even use them that often back when I still worked predominantly with C++. I was mainly trying to restore the missing functionality without breaking anything and keep to the existing code. Probably wasn't a good idea trying to fix stuff at midnight either.

Thanks for accepting and for tidying it with the new template.